WordPress/gutenberg
# Move Posts Per Page, Offset, and Pages controls from the block toolbar into Inspector Controls

## What?
This PR moves the Posts Per Page, Offset, and Max Pages to Show controls into the Inspector controls side bar and wraps them in the new ToolsPanel component.

## Why?
Having controls in multiple locations (toolbar and sidebar) is bad UX and makes these controls easy to miss. I wrote a whole plugin around adding these in 😳 .

## Testing Instructions
1. Insert a new instance of the Query Loop block
2. Confirm that there is a a new Display section in the sidebar.
3. Confirm that Posts Per Page is displayed by default.
4. Confirm that this section does not appear when the block is set to inherit the query

## Screenshots or screencast 

![screenshot](https://github.com/WordPress/gutenberg/assets/1259027/3bb762af-0464-47be-8563-7a9a46128c85)
